tst/com/amazon/corretto/crypto/provider/test/AesKeyWrapTest.java

```diff
@@ -48,41 +48,42 @@
 public final class AesKeyWrapTest {
 
     public static List<Arguments> sizes() {
-        final int[] keySizes = {16, 24, 32};
+        final int[] aeskeySizes = {16, 24, 32};
         final int[] secretSizes = {
-            8,                      // https://datatracker.ietf.org/doc/html/rfc5649#section-4.1
-            16, 24, 32,             // AES keys
-            512, 1024, 2048, 4096,  // RSA keys
-            123, 900, 81, 99, 37,   // weird sizes to exercise padding logic
+            8,                          // https://datatracker.ietf.org/doc/html/rfc5649#section-4.1
+            16, 24, 32,                 // AES keys
+            512, 1024, 2048, 4096,      // RSA keys
+            4, 123, 900, 81, 99, 37,    // weird sizes to exercise padding logic
         };
         List<Arguments> args = new ArrayList<>();
-        for (int keySize : keySizes) {
+        for (int aeskeySize : aeskeySizes) {
             for (int secretSize : secretSizes) {
-                args.add(Arguments.of(keySize, secretSize));
+                args.add(Arguments.of(aeskeySize, secretSize > 0 ? secretSize-1 : secretSize));
+                args.add(Arguments.of(aeskeySize, secretSize));
+                args.add(Arguments.of(aeskeySize, secretSize+1));
             }
         }
         return args;
     }
 
     @ParameterizedTest
     @MethodSource("sizes")
-    public void roundtrip(int keySize, int secretSize) throws Exception {
+    public void roundtrip(int aeskeySize, int secretSize) throws Exception {
```

Comment:
Wonderful!

Now, make sure you have some cases which actually wrap and unwrap real RSA and EC keys. You can probably do something like this:

```java
private void roundTripAsymmetric(SecretKey wrappingKey, KeyPairGenerator kg, KeyFactory kf) {
    KeyPair pair = kg.generate();
    Cipher c = Cipher.getInstance("AES/KWP/NoPadding", TestUtil.NATIVE_PROVIDER);
    c.init(Cipher.WRAP_MODE, wrappingKey);
    byte[] wrappedPrivateKey = c.wrap(pair.getPrivate());
    byte[] wrappedPublicKey = c.wrap(pair.getPublic());

    // Compatibility checks with the OpenJDK and BC???

    c.init(Cipher.UNWRAP_MODE, wrappingKey);
    Key unwrappedPrivate = c.unwrap(wrappedPublic, pair.getPrivate().getAlgorithm(), Cipher.PRIVATE_KEY);
    Key unwrappedPublic = c.unwrap(wrappedPublic, pair.getPublic().getAlgorithm(), Cipher.PUBLIC_KEY);

    // By passing it through the factory we ensure that it is an understandable type
    assertHexArrayEquals(kf.getKeySpec(pair.getPrivate(), PKCS8EncodedKeySpec.class).getEncoded(), kf.getKeySpec(unwrappedPrivate , X509EncodedKeySpec.class).getEncoded());
    assertHexArrayEquals(kf.getKeySpec(pair.getPublic(), X509EncodedKeySpec.class).getEncoded(), kf.getKeySpec(unwrappedPublic , X509EncodedKeySpec.class).getEncoded());

    // Compatibility checks with the OpenJDK and BC???
}
```

Reply:
will do, thank you. i have the compatibility tests with BC working locally across "generic" keys, am going to also work on parameterizing across:

- JCE compatibility for [Java >=17](https://bugs.openjdk.org/browse/JDK-8248268)
- RSA secrets across valid key sizes
- EC secrets across valid key sizes